common/test/metrics/DurationMetricTest.scala

```diff
@@ -44,4 +45,44 @@ class DurationMetricTest extends FlatSpec with Matchers {
     dataPoints.length should be(7)
     dataPoints.splitAt(4)._1 should be(allMetrics)
   }
+
+  "SamplerMetric" should "start off empty" in {
```

Comment:
The SamplerMetric functionality is almost identical to DurationMetric count, but the recording of the metric/sample is handled by 2 different functions. These tests cover the SamplerMetric recording functionality, they were missing before and helped me try to debug what was going on